client/src/modules/inventory/list/list.js

```diff
@@ -138,20 +140,30 @@ function InventoryListController($translate, Inventory, Notify, uiGridConstants,
   }
 
   function runResearch(params) {
+
     if (params) {
       search.assignFilters(params);
       vm.latestViewFilters = search.latestViewFilters();
       vm.hasCustomFilters = search.hasCustomFilters();
       vm.parameters = JSON.stringify(params);
     }
 
+    vm.loading = true;
+    vm.hasError = false;
+
     Inventory.read(null, params)
       .then(function (rows) {
         vm.gridOptions.data = rows;
       })
-      .catch(Notify.handleError);
+      .catch(Error);
+
+    vm.loading = false;
```

Comment:
Does the loading indicator work?  It seems like this will not wait for the data to download.

The correct way to wait for the data to download is to use the .finally() Promise method.  Like this:
```js
.catch(Notify.handleError)
.finally(function () {
  vm.loading = false; // this will execute after the data is downloaded.
});
```